play-akka-http-ws-standalone/src/main/scala/play/api/libs/ws/akkahttp/StandaloneAkkaHttpWSRequest.scala

```diff
@@ -0,0 +1,302 @@
+   * Adds a filter to the request that can transform the request for subsequent filters.
+   */
+  override def withRequestFilter(filter: WSRequestFilter): Self =
+    copy(filters = filters :+ filter)
+
+  /**
+   * Sets the virtual host to use in this request
+   */
+  override def withVirtualHost(vh: String): Self =
+    copy(request = request.addHeader(Host(vh)))
+
+  /**
+   * Sets the proxy server to use in this request
+   */
+  override def withProxyServer(proxyServer: WSProxyServer): Self = ???
+
+  /**
+   * Sets the method for this request
+   */
+  override def withMethod(method: String): Self =
+    copy(request = request.withMethod(
+      HttpMethods.getForKey(method)
+        .getOrElse(throw new IllegalArgumentException(s"Unknown HTTP method $method"))))
+
+  /**
+   * Sets the body for this request.
+   */
+  override def withBody[T: BodyWritable](body: T): Self = {
+    val writable = implicitly[BodyWritable[T]]
+
+    val requestWithEntity = request.withEntity(writable.transform(body) match {
+      case InMemoryBody(bytes) => HttpEntity(bytes)
+      case SourceBody(source) => HttpEntity(ContentType.parse(writable.contentType).right.get, source)
+      case EmptyBody => HttpEntity.Empty
+    })
+
+    copy(request = requestWithEntity)
+  }
+
+  /**
+   * Performs a GET.
+   */
+  override def get(): Future[Response] =
+    execute("GET")
+
+  /**
+   * Performs a PATCH request.
+   *
+   * @param body the payload body submitted with this request
+   * @return a future with the response for the PATCH request
+   */
+  override def patch[T: BodyWritable](body: T): Future[Response] =
+    withBody(body).execute("PATCH")
+
+  /**
+   * Performs a POST request.
+   *
+   * @param body the payload body submitted with this request
+   * @return a future with the response for the POST request
+   */
+  override def post[T: BodyWritable](body: T): Future[Response] =
+    withBody(body).execute("POST")
+
+  /**
+   * Performs a PUT request.
+   *
+   * @param body the payload body submitted with this request
+   * @return a future with the response for the PUT request
+   */
+  override def put[T: BodyWritable](body: T): Future[Response] =
+    withBody(body).execute("PUT")
+
+  /**
+   * Perform a DELETE on the request asynchronously.
+   */
+  override def delete(): Future[Response] =
+    execute("DELETE")
+
+  /**
+   * Perform a HEAD on the request asynchronously.
+   */
+  override def head(): Future[Response] =
+    execute("HEAD")
+
+  /**
+   * Perform a OPTIONS on the request asynchronously.
+   */
+  override def options(): Future[Response] =
+    execute("OPTIONS")
+
+  /**
+   * Executes the given HTTP method.
+   *
+   * @param method the HTTP method that will be executed
+   * @return a future with the response for this request
+   */
+  override def execute(method: String): Future[Response] =
+    withMethod(method).execute()
+
+  /**
+   * Execute this request
+   */
+  override def execute(): Future[Response] = {
```

Comment:
Good to ask play team about semantics here.
Does `execute` mean strict and `stream` "streaming" as our default mode is?
If so, then we should rather implement the `stream()` and the `execute()` should be implemented in terms of it, by applying a `toStrict()`?

Reply:
> Does execute mean strict and stream "streaming" as our default mode is?

Yes. We should make this API clearer later.

Reply:
Yes. That makes sense.